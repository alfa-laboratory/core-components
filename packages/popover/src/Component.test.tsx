import React, { ReactNode } from 'react';
import { render, act, RenderResult } from '@testing-library/react';

import { Popover, PopoverProps } from './index';

const renderPopover = async (
    props: PopoverProps & { children: ReactNode },
): Promise<RenderResult> => {
    let result: RenderResult;

    await act(async () => {
        result = await render(<Popover {...props}>{props.children}</Popover>);
    });

    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    return result;
};

describe('Display tests', () => {
    it('should display correctly', async () => {
        const component = await renderPopover({
            children: <div>I am popover</div>,
            open: true,
            anchorElement: document.body,
        });

        expect(component).toMatchSnapshot();
    });
});

describe('Attributes tests', () => {
    it('should set data-test-id attribute', async () => {
        const testId = 'tooltip';
        const className = 'custom';

        await renderPopover({
            children: <div>I am popover</div>,
            open: true,
            anchorElement: document.body,
            dataTestId: testId,
            popperClassName: className,
        });

        const popperContentWrap = document.querySelector(`.${className}`);

        const testIdAttr = popperContentWrap?.getAttribute('data-test-id');

        expect(testIdAttr).toBe(testId);
    });
});

describe('Styles tests', () => {
    it('should set custom css class to content component', async () => {
        const testId = 'tooltip';
        const className = 'custom';

        const { getByTestId } = await renderPopover({
            children: <div>I am popover</div>,
            open: true,
            anchorElement: document.body,
            dataTestId: testId,
            popperClassName: className,
        });

        expect(getByTestId(testId)).toHaveClass(className);
    });

    it('should set custom css class to arrow component', async () => {
        const arrowClassName = 'custom-arrow-class';
        const testId = 'tooltip';

        await renderPopover({
            children: <div>I am popover</div>,
            open: true,
            anchorElement: document.body,
            dataTestId: testId,
            arrowClassName,
            withArrow: true,
        });

        expect(document.body.querySelector(`.${arrowClassName}`)).toHaveClass('arrow');
    });
});

describe('Render tests', () => {
    it('should render without Transition of withTransition is false', async () => {
        const component = await renderPopover({
            children: <div>I am popover</div>,
            open: true,
            anchorElement: document.body,
            withTransition: false,
        });

        expect(component).toMatchSnapshot();
    });

    it('should unmount without errors', async () => {
        const { unmount } = await renderPopover({
            children: <div>I am popover</div>,
            open: true,
            anchorElement: document.body,
        });

        expect(unmount).not.toThrowError();
    });
});
