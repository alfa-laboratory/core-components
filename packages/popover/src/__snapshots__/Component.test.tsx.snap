// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display tests should display correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div
      alfa-portal-container=""
    >
      <div
        class="component entered entered"
        data-popper-escaped="true"
        data-popper-placement="left"
        data-popper-reference-hidden="true"
        style="position: absolute; left: 0px; top: 0px; transition-duration: 300ms; right: 0px; transform: translate(0px, 0px);"
      >
        <div>
          I am popover
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render tests should render without Transition of withTransition is false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      alfa-portal-container=""
    >
      <div
        class="component"
        data-popper-escaped="true"
        data-popper-placement="left"
        data-popper-reference-hidden="true"
        style="position: absolute; left: 0px; top: 0px; right: 0px; transform: translate(0px, 0px);"
      >
        <div>
          I am popover
        </div>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
